sequencer/src/block/queryable.rs

```diff
@@ -74,7 +74,10 @@ impl QueryablePayload for Payload<TxTableEntryWord> {
             return None; // error: index out of bounds
         }
 
-        let vid = test_vid_factory(); // TODO temporary VID construction
+        // TODO temporary VID construction. We need to get the number of storage nodes from the VID
+        // common data. May need the query service to pass common into this function along with
+        // metadata.
```

Comment:
Ugh. Shall we add a field `num_storage_nodes` to `Self::Metadata`?

Reply:
I don't know if it really makes sense to add it to metadata. Currently metadata is a byproduct of constructing the block, nothing to do with VID. If anything I think we could add it to the block header, currently `Header::new` takes `vid_commitment`. We could potentially add `num_storage_nodes` there. I also don't think it's unreasonable to have the query service pass in the common data when this function is called.

For now, though, I think we leave it as is until transaction proofs becomes a more pressing need.

Reply:
https://github.com/EspressoSystems/espresso-sequencer/issues/1155